data-prepper-core/src/main/java/org/opensearch/dataprepper/acknowledgements/DefaultAcknowledgementSet.java

```diff
@@ -84,6 +86,8 @@ public boolean isDone() {
             if (Instant.now().isAfter(expiryTime)) {
                 if (callbackFuture != null) {
                     callbackFuture.cancel(true);
+                    callbackFuture = null;
+                    LOG.warn("AcknowledgementSet expired");
```

Comment:
I am expecting there shouldn't be a lot of these. Since we only support one source, the acknowledgement set and expiry time should be obvious. We could pass a name to `AcknowledgementSetManager.create` when we support multiple sources.